src/core/mac/mac_frame.cpp

```diff
@@ -1447,14 +1447,15 @@ Frame::InfoString Frame::ToInfoString(void) const
 
 BeaconPayload::InfoString BeaconPayload::ToInfoString(void) const
 {
-    NetworkName name;
-    InfoString  string;
+    NetworkName  name;
+    InfoString   string;
```

Comment:
Why don't we initialize `string` in its constructor and make `StringWriter` just append characters not caring the string is empty or not?
You mentioned that this was your preference. @abtink

Reply:
Copying from the issue:

> I do like this as well sine it is a safer model, but did not go for it for now due to its code size impact. Adding a constructor to clear the String adds up to ~110 bytes (please see #6582). It somewhat defeats one of the main benefits of StringWriter (which is/was to reduce code size). For now, I think would be good to fix this quickly and then later we can consider different options.

Reply:
Thanks for clarafication.